Lib/_pyrepl/base_eventqueue.py

```diff
@@ -69,13 +69,19 @@ def insert(self, event: Event) -> None:
         trace('added event {event}', event=event)
         self.events.append(event)
 
-    def push(self, char: int | bytes) -> None:
+    def push(self, char: int | bytes | str) -> None:
         """
         Processes a character by updating the buffer and handling special key mappings.
         """
         ord_char = char if isinstance(char, int) else ord(char)
-        char = bytes(bytearray((ord_char,)))
-        self.buf.append(ord_char)
+        if ord_char > 255:
+            assert isinstance(char, str)
+            char = bytes(char.encode(self.encoding))
```

Comment:
In `WindowsConsole`, `encode()` is called with the `"replace"` error handler. It seems that would be best here, too -- even if the encoding is always `'utf-8'` (according to `ENCODING` in `readline.py`.)
```suggestion
            char = bytes(char.encode(self.encoding, "replace"))
```

Reply:
Thanks!